### PR TITLE
fix: removed special characters from backup file name [AR-3415]

### PR DESCRIPTION
**Pull request by Garzas**
----
#### PR Submission Checklist for internal contributors

- The **PR Title**
  - [x] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²
  - [x] contains a reference JIRA issue number like `SQPIT-764`
  - [x] answers the question: _If merged, this PR will: ..._ ³

- The **PR Description**
  - [ ] is free of optional paragraphs and you have filled the relevant parts to the best of your ability
----

# What's new in this PR?

### Issues

Backup file name was containing special character `:` from timestamp.

### Causes (Optional)

Some of file managers were not able to save files with special characters

### Solutions

Remove special character from file name

----
#### PR Post Submission Checklist for internal contributors (Optional)

 - [ ] Wire's Github Workflow has automatically linked the PR to a JIRA issue
----
#### PR Post Merge Checklist for internal contributors

 - [ ] If any soft of configuration variable was introduced by this PR, it has been added to the relevant documents and the CI jobs have been updated.
----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.

### CODE REVIEW

File: logic/src/commonMain/kotlin/com/wire/kalium/logic/feature/backup/BackupConstants.kt [hunk 1]
```diff
@@ -30,7 +30,8 @@ object BackupConstants {
     const val BACKUP_WEB_EVENTS_FILE_NAME = "events.json"
     const val BACKUP_WEB_CONVERSATIONS_FILE_NAME = "conversations.json"
 
-    fun createBackupFileName(userHandle: String?, timestampIso: String) = "$BACKUP_FILE_NAME_PREFIX-$userHandle-$timestampIso.zip"
+    fun createBackupFileName(userHandle: String?, timestampIso: String) = // file names cannot have special characters
+        "$BACKUP_FILE_NAME_PREFIX-$userHandle-${timestampIso.replace(":", "-")}.zip"
```
Comment (ohassine) [inline logic/src/commonMain/kotlin/com/wire/kalium/logic/feature/backup/BackupConstants.kt:34]: Good catch, can you cover it with test ?

**Review by saleniuk — APPROVED** [review]

File: logic/src/commonTest/kotlin/com/wire/kalium/logic/feature/backup/BackupConstantsTest.kt [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+        println(backupFileName)
+
+        // Then
+        assertTrue(isValidFileName(backupFileName))
+    }
+
+    private fun isValidFileName(filename: String): Boolean {
+        val regex = """^[a-zA-Z0-9_-]+\.[a-zA-Z0-9_]+$""".toRegex()
+        return regex.matches(filename)
+    }
```
Comment (ohassine) [inline logic/src/commonTest/kotlin/com/wire/kalium/logic/feature/backup/BackupConstantsTest.kt:46]: We might need to test `isValidFileName ` in this case.. Maybe we go easier by just asserting string values are the same

```
val exptectedValue = "$BACKUP_FILE_NAME_PREFIX-$userHandle-timestamp.zip"
assertEqual(exptectedValue, backupFileName)
```

File: logic/src/commonTest/kotlin/com/wire/kalium/logic/feature/backup/BackupConstantsTest.kt [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+
+        // Then
+        assertTrue(isValidFileName(backupFileName))
+        assertEquals(expectedValue, backupFileName)
+    }
+
+    private fun isValidFileName(filename: String): Boolean {
+        val regex = """^[a-zA-Z0-9_-]+\.[a-zA-Z0-9_]+$""".toRegex()
+        return regex.matches(filename)
+    }
```
Comment (vitorhugods) [inline logic/src/commonTest/kotlin/com/wire/kalium/logic/feature/backup/BackupConstantsTest.kt:51]: The test case is testing way more than just replacing colons with hyphens. It tests the whole backup name, and it comes up with `isValidFileName` logic, which doesn't exist in the actual code.

Also, the test name is a bit off.

I suggest keeping the test focused on what is being done in this PR:

```suggestion
    @Test
    fun givenTimeStampWithColons_whenCreatingFileNameForBackup_thenShouldReplaceColons() = runTest() {
        // Given
        val timestamp = "10:20:40.000"
        val correctedTimestamp = "10-20-40.000"
        val userHandle = "user_handle"

        val expectedValue = "$BACKUP_FILE_NAME_PREFIX-$userHandle-$correctedTimestamp.zip"

        // When
        val backupFileName = BackupConstants.createBackupFileName(userHandle, timestamp)

        // Then
        assertFalse(backupFileName.contains(":"))
        assertTrue(backupFileName.contains(correctedTimestamp))
    }
```

The assertions are now scoped to the colon/hyphen feature.

**Review by vitorhugods — CHANGES_REQUESTED** [review]
Test can be improved a bit in my opinion. It's too generic at the moment.

**Review by vitorhugods — APPROVED** [review]
👨‍🍳🤌🏻

File: logic/src/commonMain/kotlin/com/wire/kalium/logic/feature/backup/BackupConstants.kt (hunk 1 above)
Comment (gongracr) [reply logic/src/commonMain/kotlin/com/wire/kalium/logic/feature/backup/BackupConstants.kt:34]: Shall we simply remove the colon?

**Review by gongracr — APPROVED** [review]
Good catch 🐛 🦅

**Pull request merged**